packages/google_maps_flutter/google_maps_flutter/ios/Classes/GoogleMapPolygonController.m

```diff
@@ -56,156 +60,156 @@ - (void)setHoles:(NSArray<NSArray<CLLocation *> *> *)rawHoles {
+                                         registrar:self.registrar];
+    self.polygonIdentifierToController[identifier] = controller;
   }
 }
+
 - (void)changePolygons:(NSArray *)polygonsToChange {
   for (NSDictionary *polygon in polygonsToChange) {
-    NSString *polygonId = [FLTPolygonsController getPolygonId:polygon];
-    FLTGoogleMapPolygonController *controller = _polygonIdToController[polygonId];
+    NSString *identifier = [FLTPolygonsController getPolygonId:polygon];
+    FLTGoogleMapPolygonController *controller = self.polygonIdentifierToController[identifier];
     if (!controller) {
       continue;
     }
-    InterpretPolygonOptions(polygon, controller, _registrar);
+    [FLTPolygonsController interpretPolygonOptions:polygon
+                                              sink:controller
+                                         registrar:self.registrar];
   }
 }
-- (void)removePolygonIds:(NSArray *)polygonIdsToRemove {
-  for (NSString *polygonId in polygonIdsToRemove) {
-    if (!polygonId) {
-      continue;
-    }
-    FLTGoogleMapPolygonController *controller = _polygonIdToController[polygonId];
+
+- (void)removePolygonWithIdentifiers:(NSArray *)identifiers {
+  for (NSString *identifier in identifiers) {
+    FLTGoogleMapPolygonController *controller = self.polygonIdentifierToController[identifier];
     if (!controller) {
       continue;
     }
     [controller removePolygon];
-    [_polygonIdToController removeObjectForKey:polygonId];
+    [self.polygonIdentifierToController removeObjectForKey:identifier];
   }
 }
-- (void)onPolygonTap:(NSString *)polygonId {
-  if (!polygonId) {
+
+- (void)didTapPolygonWithIdentifier:(NSString *)identifier {
+  if (!identifier) {
     return;
   }
-  FLTGoogleMapPolygonController *controller = _polygonIdToController[polygonId];
+  FLTGoogleMapPolygonController *controller = self.polygonIdentifierToController[identifier];
   if (!controller) {
     return;
   }
-  [_methodChannel invokeMethod:@"polygon#onTap" arguments:@{@"polygonId" : polygonId}];
+  [self.methodChannel invokeMethod:@"polygon#onTap" arguments:@{@"polygonId" : identifier}];
 }
-- (bool)hasPolygonWithId:(NSString *)polygonId {
-  if (!polygonId) {
+
+- (bool)hasPolygonWithIdentifier:(NSString *)identifier {
+  if (!identifier) {
     return false;
   }
-  return _polygonIdToController[polygonId] != nil;
+  return self.polygonIdentifierToController[identifier] != nil;
 }
+
 + (GMSMutablePath *)getPath:(NSDictionary *)polygon {
   NSArray *pointArray = polygon[@"points"];
-  NSArray<CLLocation *> *points = ToPoints(pointArray);
+  NSArray<CLLocation *> *points = [FLTGoogleMapJSONConversions pointsFromLatlongs:pointArray];
   GMSMutablePath *path = [GMSMutablePath path];
   for (CLLocation *location in points) {
     [path addCoordinate:location.coordinate];
   }
   return path;
 }
+
 + (NSString *)getPolygonId:(NSDictionary *)polygon {
   return polygon[@"polygonId"];
 }
+
++ (void)interpretPolygonOptions:(NSDictionary *)data
```

Comment:
Same thing, can we get rid of `FLTGoogleMapPolygonOptionsSink` and make these properties on `FLTGoogleMapPolygonController`?